pkg/cloudclient/aws/aws.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sts"
+
+	"github.com/golang/glog"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	awsclient "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/aws/client"
+	ccerrors "github.com/openshift/configuration-anomaly-detection/pkg/cloudclient/errors"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/pkg/errors"
+)
+
+const (
+	jumpRole      = "ba"
+	defaultRegion = "us-east-1" // TODO extract to own config
+	credentialDir = "~/.aws"    // TODO extract to own config
+)
+
+const (
+	// the ClusterDeployment label key we're expecting to be able to pull the ClusterID from
+	cdClusterIDLabelKey string = "api.openshift.com/id"
+)
+
+// CloudClient is the client used for handling AWS Cloud Handler requests
+type CloudClient struct {
+	// clusterDeployment is the ClusterDeployment for the target cluster (the one
+	// for which a login url is requested)
+	clusterDeployment *hivev1.ClusterDeployment
+
+	// awsClient is the awsClient that has the BACKPLANE user credentials.  This client
+	// is used to create other clients in order to perform the necessary role chaining to
+	// access a customer's account via STS assume roles.
+	awsClient awsclient.AwsClient
```

Comment:
Why call the struct awsclient?
It's in a separate package

Reply:
not sure what's the problem... 
But I guess the fact that there is 3x awsClient, once used for the struct, the package, and the Interface. Yes, that can be improved.
Since we need to create several clients on-demand with different credentials, it makes sense to pack the client inside of its own package.
This variable holds the client to the origin of the chain. We may want to change that to the target, meaning the customer account in the end. Then this would become the client for executing commands in the customer AWS account

Reply:
I meant the package can be called awsClient, the struct in the awsClient can be called Client and the field could be client or awsClient